packages/snap-preact/src/setupEvents.test.tsx

```diff
@@ -193,4 +207,67 @@ describe('setupEvents', () => {
 			jest.clearAllMocks();
 		});
 	});
+
+	describe('controller/updateRecs', () => {
+		it('can listen for controller/updateRecs event and invoked without parameters', () => {
+			expect(makeSearchSpy).not.toHaveBeenCalled();
+
+			//no data passed at all
+			expect(() => {
+				eventManager.fire('controller/updateRecs');
+			}).not.toThrow();
+
+			expect(makeSearchSpy).toHaveBeenCalledTimes(1);
+
+			jest.clearAllMocks();
+		});
+
+		it('can pass exact match controllerId', () => {
+			expect(makeSearchSpy).not.toHaveBeenCalled();
+			const controllerIds = 'recommend_similar_0';
+			expect(() => {
+				eventManager.fire('controller/updateRecs', { controllerIds });
+			}).not.toThrow();
+
+			expect(makeSearchSpy).toHaveBeenCalledTimes(1);
+
+			jest.clearAllMocks();
+		});
+
+		it('can pass partial regex match controllerId', () => {
+			expect(makeSearchSpy).not.toHaveBeenCalled();
+			const controllerIds = [/^recommend_/];
+			expect(() => {
+				eventManager.fire('controller/updateRecs', { controllerIds });
+			}).not.toThrow();
+
+			expect(makeSearchSpy).toHaveBeenCalledTimes(1);
+
+			jest.clearAllMocks();
+		});
+
+		it('only invokes for recommendation type and if exists', () => {
+			expect(makeSearchSpy).not.toHaveBeenCalled();
+			const controllerIds = ['recommend_similar_0', 'search', 'dne'];
+			expect(() => {
+				eventManager.fire('controller/updateRecs', { controllerIds });
+			}).not.toThrow();
+
+			expect(makeSearchSpy).toHaveBeenCalledTimes(1);
```

Comment:
Again test should verify the right controller search functions were called for provided `id` values.